src/popup.tsx

```diff
@@ -1,198 +1,58 @@
-              {(params: { address: string }) => (
-                <ContactSettings address={params?.address} />
-              )}
-            </Route>
-          </Switch>
-          <Route
-            path="/quick-settings/notifications"
-            component={NotificationSettings}
-          />
-          <Route path="/subscriptions/:id">
-            {(params: { id: string }) => (
-              <SubscriptionDetails id={params?.id} />
-            )}
-          </Route>
-          <Route path="/subscriptions/:id/manage">
-            {(params: { id: string }) => (
-              <SubscriptionManagement id={params?.id} />
-            )}
-          </Route>
-          <Route path="/subscriptions/:id/payment">
-            {(params: { id: string }) => (
-              <SubscriptionPayment id={params?.id} />
-            )}
-          </Route>
-          <Route path="/transactions" component={Transactions} />
-          <Route path="/notifications" component={Notifications} />
-          <Route path="/notification/:id">
-            {(params: { id: string }) => (
-              <MessageNotification id={params?.id} />
-            )}
-          </Route>
-          <Route path="/tokens" component={Tokens} />
-          <Route path="/token/:id">
-            {(params: { id: string }) => <Asset id={params?.id} />}
-          </Route>
-          <Route path="/collectibles" component={Collectibles} />
-          <Route path="/collectible/:id">
-            {(params: { id: string }) => <Collectible id={params?.id} />}
-          </Route>
-          <Route path="/transaction/:id/:gateway?">
-            {(params: { id: string; gateway?: string }) => (
-              <Transaction id={params?.id} gw={params?.gateway} />
-            )}
-          </Route>
-          <Route path="/send/confirm/:token/:qty/:recipient/:message?">
-            {(params: { token: string; qty: string }) => (
-              <Confirm
-                tokenID={params?.token}
-                qty={Number(params?.qty || "0")}
-              />
-            )}
-          </Route>
-          <Route path="/send/recipient/:token/:qty/:message?">
-            {(params: { token: string; qty: string; message?: string }) => (
-              <Recipient
-                tokenID={params?.token}
-                qty={params?.qty || "0"}
-                message={params?.message || ""}
-              />
-            )}
-          </Route>
-          <NavigationBar />
-        </HistoryProvider>
-      </Router>
+      <>
+        <Routes routes={POPUP_ROUTES} />
+        <NavigationBar />
+      </>
     );
   }
 
+  return <>{content}</>;
+}
+
+export function ArConnectBrowserExtensionAppRoot() {
+  const initialScreenType = useBrowserExtensionWalletSetUp();
+
   return (
     <ArConnectThemeProvider>
-      <AnimatePresence initial={false}>{content}</AnimatePresence>
+      <Wouter hook={useHashLocation}>
+        <BodyScroller />
+        <HistoryObserver />
```

Comment:
TODO: Test if `HistoryObserver` and `HistoryProvider` are actually needed or if they can be removed.